plexpy/api.py

```diff
@@ -13,86 +16,134 @@
 #  You should have received a copy of the GNU General Public License
 #  along with PlexPy.  If not, see <http://www.gnu.org/licenses/>.
 
-from plexpy import db, cache, versioncheck, logger, helpers
-
+from plexpy import versioncheck, logger, plextv, pmsconnect, datafactory, graphs, users
+import os
 import plexpy
 import json
-from xml.dom import minidom
+import traceback
+import cherrypy
+import re
+import hashlib
+import random
+import xmltodict
 
-cmd_list = ['getHistory', 'getLogs', 'getVersion', 'checkGithub', 'shutdown', 'restart', 'update']
+cmd_list = ['getLogs', 'getVersion', 'checkGithub', 'shutdown', 'getSettings',
+            'restart', 'update', 'getApikey', 'getHistory', 'getMetadata', 'getUserips',
+            'getPlayby']
 
 
 class Api(object):
-
-    def __init__(self):
+    def __init__(self, out='json'):
 
         self.apikey = None
         self.cmd = None
         self.id = None
-
         self.kwargs = None
-
         self.data = None
-
+        self.msg = None
         self.callback = None
+        self.authenticated = False
+        self.out_type = out
+        self.result_type = 'error'
 
     def checkParams(self, *args, **kwargs):
 
         if not plexpy.CONFIG.API_ENABLED:
-            self.data = 'API not enabled'
-            return
+            self.msg = 'API not enabled'
         if not plexpy.CONFIG.API_KEY:
```

Comment:
All the following if statements should either be `elif`s or this function should return something on each evaluation otherwise these `if`s are just ignored.